opensrp-giz-malawi/src/main/java/org/smartregister/giz/repository/GizEventRepository.java

```diff
@@ -21,4 +27,34 @@ public boolean hasEvent(@NonNull String baseEntityId, @NonNull String eventType)
         }
         return hasEvent;
     }
+
+    public void processSkippedClients() {
+        String missingClientRegister = "SELECT DISTINCT baseEntityId, formSubmissionId AS formSubmissionId FROM event WHERE baseEntityId IN (SELECT baseEntityId FROM client WHERE baseEntityId NOT IN (SELECT DISTINCT base_entity_id FROM client_register_type)) AND eventType LIKE '%Registration%'";
+        String missingECClient = "SELECT DISTINCT baseEntityId, formSubmissionId AS formSubmissionId FROM event WHERE baseEntityId IN (SELECT baseEntityId FROM client WHERE baseEntityId NOT IN (SELECT DISTINCT base_entity_id FROM ec_client)) AND eventType LIKE '%Registration%'";
+        List<String> formSubmissionIDs = new ArrayList<>();
+        addFormSubmissionIds(formSubmissionIDs, missingClientRegister);
+        addFormSubmissionIds(formSubmissionIDs, missingECClient);
+
+        try {
+            if (formSubmissionIDs.size() > 0)
+                GizUtils.initiateEventProcessing(formSubmissionIDs);
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+    }
+
+
+    public void addFormSubmissionIds(List<String> ids, String query) {
+        SQLiteDatabase database = getReadableDatabase();
+        Cursor cursor = database.rawQuery(query, null);
+        int columnIndex = cursor.getColumnIndex("formSubmissionId");
+        if (cursor.getCount() > 0) {
+            cursor.moveToFirst();
+            do {
+                if (!ids.contains(cursor.getString(columnIndex)))
```

Comment:
Time complexity for contains is n. I still believe that a HashSet is better here